earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -247,90 +245,70 @@ void PacketGenAssetVersionImplD::UpdateChildren(
       }
 
       // populate inputs and inputVers & config.insets
-      levInputs.reserve(mergeIndexes.size() * 2 + 1);
-      levInputVers.reserve(mergeIndexes.size() * 2 + 1);
       packetLevelConfig.insets.reserve(mergeIndexes.size() + 1);
-      for (std::vector<uint>::const_iterator idx = mergeIndexes.begin();
-           idx != mergeIndexes.end(); ++idx) {
-        const InsetInfo<ProductAssetVersion> &inset(*extra.insetInfo[*idx]);
-        const PacketGenInfo &genInfo(*extra.genInfo[*idx]);
-        AssetVersion blendVer;
+      packetLevelConfig.insets.resize(mergeIndexes.size());
+      int threadsToUse = MiscConfig::Instance().PackGenThreads;
+      levInputs.resize(mergeIndexes.size() * 2 + 1);
+      levInputVers.resize(mergeIndexes.size() * 2 + 1);
+      AssetVersion blendVer;
+
+      // Run the following loop in parallel using the amount of threads set in misc.xml.
+      #pragma omp parallel for schedule(static) num_threads(threadsToUse)
+      for (size_t merge_idx = 0; merge_idx < mergeIndexes.size(); ++merge_idx) {
+        const InsetInfo<ProductAssetVersion> &inset(*extra.insetInfo[mergeIndexes[merge_idx]]);
+        const PacketGenInfo &genInfo(*extra.genInfo[mergeIndexes[merge_idx]]);
+        int inputs_idx = merge_idx*2;
+        levInputs[inputs_idx] = (inset.combinedrp_).Ref();
+        levInputVers[inputs_idx] = inset.combinedrp_;
+
         if (!inset.alphaRPFile.empty() &&
             (genInfo.packetLevels.size() > minifySrcLevel)) {
           blendVer = genInfo.packetLevels[minifySrcLevel];
         }
 
-        if (!blendVer) {
-          // This inset doesn't have a cached blend file. This
+        if (blendVer) {
```

Comment:
I would suggest changing the general flow of logic back the way it was unless there's a compelling reason to keep it the way it is now. The if/else conditions are kind of backward to what they were before, and it makes it fairly difficult to compare the old and new versions and feel confident that they're doing the same thing. 
I think it would be best to keep it as
```
if (!blendVer) {
    // Do same stuff as before but with the thread-safe constructs.
}
else {
    // Do same stuff as before but with the thread-safe constructs.
    // Of course, collapse the two identical cases into one as you've already done.
}
```